worldwind/src/commonMain/kotlin/earth/worldwind/render/program/GeomLinesShaderProgram.kt

```diff
@@ -0,0 +1,216 @@
+package earth.worldwind.render.program
+
+import earth.worldwind.draw.DrawContext
+import earth.worldwind.geom.Matrix3
+import earth.worldwind.geom.Matrix4
+import earth.worldwind.geom.Vec2
+import earth.worldwind.geom.Viewport
+import earth.worldwind.render.Color
+import earth.worldwind.util.kgl.KglUniformLocation
+
+open class GeomLinesShaderProgram : AbstractShaderProgram() {
+    override var programSources = arrayOf(
+        """
+            uniform mat4 mvpMatrix;
+            uniform float lineWidth;
+            uniform vec2 screen;
+            uniform bool enableTexture;
+            uniform bool enableOneVertexMode;
+            uniform mat3 texCoordMatrix;
+            
+            attribute vec4 pointA;
+            attribute vec4 pointB;
+            attribute vec4 pointC;
+            attribute vec2 vertexTexCoord;
+            
+            varying vec2 texCoord;
+            
+            void main() {
+                if(enableOneVertexMode)
+                {
+                    /* Transform the vertex position by the modelview-projection matrix. */
+                    gl_Position = mvpMatrix * pointA;
+                }
+                else
+                {
+                    /* Transform the vertex position by the modelview-projection matrix. */
+                    vec4 pointAScreen = mvpMatrix * vec4(pointA.xyz, 1);
+                    vec4 pointBScreen = mvpMatrix * vec4(pointB.xyz, 1);
+                    vec4 pointCScreen = mvpMatrix * vec4(pointC.xyz, 1);
+                    float corner =  pointB.w;
+                    
+                    pointAScreen = pointAScreen / pointAScreen.w;
+                    pointBScreen = pointBScreen / pointBScreen.w;
+                    pointCScreen = pointCScreen / pointCScreen.w;
+                    
+                    if(all(equal(pointBScreen.xy, pointAScreen.xy)))
```

Comment:
Should be epsilon comparison

Reply:
all(lessEqual(abs(), EPS))
EPS should be based on pixel size

Reply:
done